subsys/usb/class/hid/core.c

```diff
@@ -108,9 +108,10 @@ static int hid_custom_handle_req(struct usb_setup_packet *setup,
 		switch (setup->wValue) {
 		case 0x2200:
 			SYS_LOG_DBG("Return Report Descriptor");
-			if (*len > hid_device.report_size) {
-				SYS_LOG_ERR("Trying to read too much");
-				return -ENODATA;
+			if (*len != hid_device.report_size) {
+				SYS_LOG_WRN("len %d doesn't match"
+					    "Report Descriptor size\n", *len);
+				*len = min(*len, hid_device.report_size);
 			}
```

Comment:
Not that I like this way, but it should send correct data instead of error, does win10 works well after that?
Please add comment block explaining this workaround.

Reply:
yes win10 works after that, i will add comment tomorrow.